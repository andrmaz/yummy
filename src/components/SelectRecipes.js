import React from 'react';
import PropTypes from 'prop-types';
import { Controller } from 'react-hook-form';
import { Flex, FormLabel, Select} from '@chakra-ui/react';

const SelectRecipes = ({ control }) => {
    return (
        <Controller
            name="to"
            data-testid='b'
            control={control}
            defaultValue={4}
            render={({ onChange, value }) =>
                <Flex direction='column'> 
                    <FormLabel textAlign='center' pt='5px' >Recipes</FormLabel>
                    <Select 
                        id='to'
                        onChange={onChange} 
                        value={value} 
                        name='to'
                        size='xs' 
                        w='200px'
                        m='5px'
                        variant="filled"
                        focusBorderColor='blue.500'
                    >
                        <option label={2} value={2} />
                        <option label={3} value={3} />
                        <option label={4} value={4} />
                        <option label={5} value={5} />
                        <option label={6} value={6} />
                    </Select>
                </Flex>
            }
        />
    );
};

SelectRecipes.propTypes = {
    control: PropTypes.object,
};

export default SelectRecipes;
