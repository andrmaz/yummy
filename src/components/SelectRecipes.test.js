import React from 'react';
import { shallow } from 'enzyme';
import SelectRecipes from './SelectRecipes';

let wrapper;

describe('SelectRecipes Component', () => {
    beforeEach(() => {
        expect.hasAssertions();
    });
    it('should render Component correctly', () => {
        wrapper = shallow(<SelectRecipes />);
        expect(wrapper).toMatchSnapshot();
    });
});