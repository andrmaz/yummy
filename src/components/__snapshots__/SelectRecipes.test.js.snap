// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectRecipes Component should render Component correctly 1`] = `
<Controller
  data-testid="b"
  defaultValue={4}
  name="to"
  render={[Function]}
/>
`;
